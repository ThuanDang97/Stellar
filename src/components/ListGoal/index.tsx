import React, { memo } from 'react'
import { useTheme } from 'styled-components'

// Components
import CardGoal from '@components/CardGoal'
import { ViewStyled } from '@components/styled-components/ViewStyled'

// Constants
import { LIST_GOAL } from '@constants/index'

// Props type
import {
  GoalProps,
  SelectItemProps,
} from '@self-types/components/CardGold.props'

// Utils
import { findItemArrStringByValue, selectItem } from '@utils/index'

const ListGoal = ({
  selectedItem,
  setSelectedItem,
  selectedGoals,
}: SelectItemProps) => {
  const theme = useTheme()

  return (
    <ViewStyled
      display="flex"
      flexWrap="wrap"
      justifyContent="center"
      width={theme.metrics.width.full}
    >
      {LIST_GOAL.map((goal: GoalProps) => {
        const checkGoalIdIsExit =
          selectedGoals &&
          findItemArrStringByValue({
            data: selectedGoals,
            value: goal.id,
          })
        const handleSelectGoal = () =>
          !checkGoalIdIsExit &&
          selectItem(setSelectedItem, selectedItem, goal.id)

        const isActive = selectedItem.includes(goal.id)
        return (
          <CardGoal
            key={`{goal-${goal.id}`}
            title={isActive ? goal.activeText : goal.title}
            iconUrl={goal.url}
            isActive={isActive}
            handleSelectGoal={handleSelectGoal}
            disable={!!checkGoalIdIsExit}
          />
        )
      })}
    </ViewStyled>
  )
}

export default memo(ListGoal)
