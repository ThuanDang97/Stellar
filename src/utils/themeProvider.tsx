import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
