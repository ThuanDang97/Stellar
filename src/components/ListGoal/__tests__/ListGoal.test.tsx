import '@testing-library/jest-dom'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Components
import ListGoal from '@components/ListGoal'

describe('ListGoal Component', () => {
  const props = {
    selectedItem: ['goal-g1'],
    setSelectedItem: jest.fn(),
  }
  it('renders ListGoal component', () => {
    const { container } = renderWithTheme(
      <ListGoal
        selectedItem={props.selectedItem}
        setSelectedItem={props.setSelectedItem}
        selectedGoals={[]}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
