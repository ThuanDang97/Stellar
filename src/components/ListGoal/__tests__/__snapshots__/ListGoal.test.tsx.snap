// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGoal Component renders ListGoal component 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 etuUCN"
    display="flex"
    width="100%"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Improve My 
 Credit Score
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Build 
 Wealth
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Buy A Car
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Get A Job
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Buy 
 A Home
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Rent 
 A Home
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Get A 
 Credit Card
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Get Better 
 Insurance Rate
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Get Better 
 Interest Rates
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Pay May Bills 
 On Time
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Get A Loan
      </p>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
      cursor="pointer"
      data-testid="card-goal"
      display="flex"
      height="120"
      width="160"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Card Goal"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <p
        class="TextStyled-sc-9xia41-0 eTpfgb"
        color="#203F59"
        display="-webkit-box"
        font-size="14"
        overflow="hidden"
      >
        Start 
 A Business
      </p>
    </div>
  </div>
</div>
`;
